insights/command_parser.py

```diff
@@ -46,6 +53,9 @@ def __init__(self):
         # Use dispatch pattern to execute command
         getattr(self, args.command)()
 
+    def parse_version_arg(self):
```

Comment:
Can you help me understand the manual check in the arg list here? I see you add `--version` to the arg parser at line 40. Can you check `parser.version` after the args have been parsed?

Reply:
Yes. If `parser.parse_args()` is invoked when only `--version` flag is provided (`insights --run`) then ArgumentParser will throw an exception stating that required positional argument `command` is missing. Additionally, for version info display use cases, `ArgumentParser.add_argument` call provides `version` action which will parse everything correctly; however, older version of python have a bug where version info is printed into `stderr`. So for python >= 3.4, the info goes to stdout and for older python it goes to `stderr`. For those 2 reasons I ended up with manual check for `--version`. Does that make sense?

Reply:
And for the most part, the reason I kept `parser.add_argument('--version'...` is because it will generate correct help for the flag.